source/distributed/tria.cc

```diff
@@ -3309,7 +3303,15 @@ namespace parallel
       if (cell_attached_data.n_attached_deserialize > 0)
         {
           --cell_attached_data.n_attached_deserialize;
-          cell_attached_data.attached_data_pack_callbacks.pop_front();
```

Comment:
This is the only functional change: we just dropped the *first* in the list, which did not necessarily correspond to the data that is being deserialized. In the code below, we actually find the data set that is being deserialized and drop that one.